src/main/java/picard/nio/PicardBucketUtils.java

```diff
@@ -67,19 +65,48 @@ public static PicardHtsPath getTempFilePath(final String directory, String prefi
         }
     }
 
-    // For local temp file, directory should be null.
+    /**
+     * This overload of getTempFilePath takes the directory of type PicardHtsPath instead of String.
+     */
```

Comment:
Might as well use the standard javadoc format like you did elsewhere, i.e. list the params like this (this is a random example I took from elsewhere):

/**
* Picks a random name, by putting some random letters between "prefix" and "suffix".
*
* @param stagingLocation The folder where you want the file to be. Must start with "gs://" or "hdfs://"
* @param prefix The beginning of the file name
* @param suffix The end of the file name, e.g. ".tmp"
*/

Reply:
Right; I thought it might be a little verbose to repeat the almost identical javadoc for overladed methods. It seems like using `@see` to refer to the original with the full java may be a good solution?

https://stackoverflow.com/questions/2558695/javadoc-reuse-for-overloaded-methods

Reply:
Using the `see` notation seems fine, but please keep in mind that the ultimate goal here is to eliminate these string overloads, since they're the ones that cause problems. (In fact, I would lobby that the next cloudification PR should do exactly that - do nothing except clean up and get rid of all of these methods that take a string path or dir before we accumulate so many usages of them that removing them becomes untenable).

So given that, if we're going to use "see" references like this, it would be preferable to have the `public static PicardHtsPath getTempFilePath(final PicardHtsPath directory, String prefix, final String extension)` be the one with specific javadoc, and have the other ones `see`/refer to that one. Then when we remove the individual ones, we don't have to change anything else.

Reply:
If I'm not mistaken, what we want to avoid is directly manipulating the underlying URI String of an existing PicardHtsPath object to create a new one. Given that getTempFilePath is a method that creates a new PicardHtsPath from scratch, rather than from an existing PicardHtsPath, I think String input can be the main method signature (much like the constructor for PicardHtsPath itself).